pyccl/tests/test_power.py

```diff
@@ -170,3 +170,46 @@ def test_power_sigma8norm_norms_consistent(tf):
         ccl.linear_matter_power(cosmo, 1e-4, a) /
         ccl.linear_matter_power(cosmo_s8, 1e-4, a))
     assert np.allclose(pk_rat, gfac)
+
+
+def test_input_lin_power_spectrum():
+    cosmo = ccl.Cosmology(Omega_c=0.27, Omega_b=0.05, h=0.7, n_s=0.965,
+                          A_s=2e-9)
+    a_arr = np.linspace(0.1, 1.0, 50)
+    k_arr = np.logspace(np.log10(2e-4), np.log10(1), 1000)
+    pk_arr = np.empty(shape=(len(a_arr), len(k_arr)))
+    for i, a in enumerate(a_arr):
+        pk_arr[i] = ccl.power.linear_matter_power(cosmo, k_arr, a)
+
+    chi_from_ccl = ccl.background.comoving_radial_distance(cosmo, a_arr)
+    hoh0_from_ccl = ccl.background.h_over_h0(cosmo, a_arr)
+    growth_from_ccl = ccl.background.growth_factor(cosmo, a_arr)
+    fgrowth_from_ccl = ccl.background.growth_rate(cosmo, a_arr)
+
+    cosmo_input = ccl.Cosmology(Omega_c=0.27, Omega_b=0.05, h=0.7, n_s=0.965,
+                                A_s=2e-9)
+    cosmo_input._set_background_from_arrays(a_array=a_arr,
+                                            chi_array=chi_from_ccl,
+                                            hoh0_array=hoh0_from_ccl,
+                                            growth_array=growth_from_ccl,
+                                            fgrowth_array=fgrowth_from_ccl)
+    cosmo_input._set_linear_power_from_arrays(a_arr, k_arr, pk_arr)
+    cosmo_input.compute_linear_power()
+    pk_CCL_input = ccl.power.linear_matter_power(cosmo_input, k_arr, 0.5)
```

Comment:
I think I see what is happening here, but having to call cosmo_input.compute_linear_power() and then also call ccl.power.linear_matter_power() seems like a clunky thing to need to do which will be confusing for some users ... do we envision this as the standard way to do this is this something which is temporary while we are keeping this under the hood?

Reply:
It is not necessary to call "compute_linear_power()" for this to work, I think it was left there from my debugging. I have now removed it.